src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs

```diff
@@ -1401,16 +1398,15 @@ private void ToggleTreeViewItemHighlighting (string path, bool isSelected)
             }
 
             // Next, deselect the parentnode in HelixWatch3DView
-
             var nodeGeometryModels = Model3DDictionary.Where(x => x.Key.Contains(nodePath) && x.Value is GeometryModel3D).ToArray();
-
             foreach (var nodeGeometryModel in nodeGeometryModels)
             {
                 nodeGeometryModel.Value.SetValue(AttachedProperties.ShowSelectedProperty, false);
             }
 
             // Then, select the individual node
-            var geometryModels = Model3DDictionary.Where(x => x.Key.Contains(path) && x.Value is GeometryModel3D).ToArray();
+            var pathInDict = path + ":";
+            var geometryModels = Model3DDictionary.Where(x => x.Key.Contains(pathInDict) && x.Value is GeometryModel3D).ToArray();
 
             foreach (var geometryModel in geometryModels)
```

Comment:
I guess this loop doesn't need to get executed if we know that the boolean value `isSelected` is false. The previous loop above already marked the whole list as false, is that the case?